tests/unit/fixtures.py

```diff
@@ -0,0 +1,118 @@
+        self.mock_s3_requirer = VaultCharmFixtures.patcher_s3_requirer.start().return_value
+        self.mock_s3 = VaultCharmFixtures.patcher_s3.start()
+        self.mock_socket_fqdn = VaultCharmFixtures.patcher_socket_fqdn.start()
+        self.mock_pki_requirer_request_certificate_creation = (
+            VaultCharmFixtures.patcher_pki_requirer_request_certificate_creation.start()
+        )
+        self.mock_pki_requirer_get_assigned_certificates = (
+            VaultCharmFixtures.patcher_pki_requirer_get_assigned_certificates.start()
+        )
+        self.mock_pki_provider_get_outstanding_certificate_requests = (
+            VaultCharmFixtures.patcher_pki_provider_get_outstanding_certificate_requests.start()
+        )
+        self.mock_pki_provider_set_relation_certificate = (
+            VaultCharmFixtures.patcher_pki_provider_set_relation_certificate.start()
+        )
+        self.mock_autounseal_provides_get_outstanding_requests = (
+            VaultCharmFixtures.patcher_autounseal_provides_get_outstanding_requests.start()
+        )
+        self.mock_autounseal_provides_set_data = (
+            VaultCharmFixtures.patcher_autounseal_provides_set_data.start()
+        )
+        self.mock_autounseal_requires_get_details = (
+            VaultCharmFixtures.patcher_autounseal_requires_get_details.start()
+        )
+        self.mock_kv_provides_get_outstanding_kv_requests = (
+            VaultCharmFixtures.patcher_kv_provides_get_outstanding_kv_requests.start()
+        )
+        self.mock_kv_provides_set_ca_certificate = (
+            VaultCharmFixtures.patcher_kv_provides_set_ca_certificate.start()
+        )
+        self.mock_kv_provides_set_egress_subnets = (
+            VaultCharmFixtures.patcher_kv_provides_set_egress_subnets.start()
+        )
+        self.mock_kv_provides_set_vault_url = (
+            VaultCharmFixtures.patcher_kv_provides_set_vault_url.start()
+        )
+        self.mock_kv_provides_get_credentials = (
+            VaultCharmFixtures.patcher_kv_provides_get_credentials.start()
+        )
+        self.mock_get_binding = VaultCharmFixtures.patcher_get_binding.start()
+        self.mock_snap_cache = VaultCharmFixtures.patcher_snap_cache.start()
+        self.mock_machine = VaultCharmFixtures.patcher_machine.start().return_value
+        self.mock_get_common_name_from_csr = (
+            VaultCharmFixtures.patcher_get_common_name_from_csr.start()
+        )
+
+    @pytest.fixture(autouse=True)
+    def context(self):
+        self.ctx = scenario.Context(
+            charm_type=VaultOperatorCharm,
+        )
+
+
+class MockNetwork:
+    def __init__(self, bind_address: str, ingress_address: str):
+        self.bind_address = bind_address
+        self.ingress_address = ingress_address
+
+
+class MockBinding:
+    def __init__(self, bind_address: str, ingress_address: str):
+        self.network = MockNetwork(bind_address=bind_address, ingress_address=ingress_address)
```

Comment:
Scenario should already have built-in Network mocking support like [so](https://github.com/canonical/gocert-k8s-operator/blob/4ab6771ca20aa38fa6b80639f97d1053e5bc41dc/tests/unit/test_charm.py#L1457). In the case of vault, the relation string should be 'vault-peers' instead.